staging/src/k8s.io/apiserver/pkg/endpoints/request/requestinfo.go

```diff
@@ -238,11 +247,26 @@ func (r *RequestInfoFactory) NewRequestInfo(req *http.Request) (*RequestInfo, er
 			}
 		}
 	}
+
 	// if there's no name on the request and we thought it was a delete before, then the actual verb is deletecollection
 	if len(requestInfo.Name) == 0 && requestInfo.Verb == "delete" {
 		requestInfo.Verb = "deletecollection"
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.AuthorizeWithSelectors) {
+		if requestInfo.Verb == "list" || requestInfo.Verb == "watch" {
+			// interestingly these are parsed above, but the current structure there means that if one (or anything) in the
+			// listOptions fails to decode, the field and label selectors are lost.
```

Comment:
the current state is ~fine, but if listOptions fails to decode, won't the request fail anyway? I'm not sure I care a whole lot about authorizing as narrowly as possible if the request is going to fail, but happy to think about this more in a follow-up if we care